src/index.ts

```diff
@@ -233,73 +306,86 @@ class SvgChunkWebpackPlugin {
 		// Check if the filename option contains the placeholder [hash]
 		// [hash] corresponds to the Webpack compilation hash
 		if (REGEXP_HASH.test(filename)) {
-			filename = filename.replace('[hash]', this.getBuildHash());
+			filename = filename.replace('[hash]', compilation.hash);
 		}
 
 		// Check if the filename option contains the placeholder [chunkhash]
 		// [chunkhash] corresponds to the hash of the chunk content
 		if (REGEXP_CHUNKHASH.test(filename)) {
-			filename = filename.replace('[chunkhash]', this.getChunkHash(entryName));
+			const hash = compilation.entrypoints.get(entryName).chunks?.[0].hash || '';
```

Comment:
Yeah, by the way webpack has API for such things:
```
const newFilename = compilation.getPath(this.options.filename, {
  filename: name,
});
```

So I think it can be replaced with it, also it is cachable, I don't sure about every `[placeholder]`, but you can check it

Reply:
@alexander-akait The `getPath` function does not work with the `[contenthash]` placeholder, should I keep this generation/replacement in the plugin?

About other placeholders: 
- `[hash]` is deprecated, will be replaced by `[fullhash]` and it works with the `getPath` function
- `[chunkhash]` is not working but I think it doesn't matter, can be removed

Reply:
I see, yeah, we should improve `[contenthash]` logic, now - you need to calculate it on your side and pass it to function, so current solution is right